.github/workflows/pr-update-playwright-expectations.yaml

```
@@ -12,74 +12,167 @@ concurrency:
+      - name: Update snapshots (Shard ${{ matrix.shardIndex }}/${{ matrix.shardTotal }})
         id: playwright-tests
-        run: pnpm exec playwright test --update-snapshots
+        run: pnpm exec playwright test --update-snapshots --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
         continue-on-error: true
-      - uses: actions/upload-artifact@v4
+
+      # Upload updated snapshots from this shard
+      - name: Upload updated snapshots
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: snapshots-shard-${{ matrix.shardIndex }}
+          path: browser_tests/**/*-snapshots/
+          retention-days: 1
+
+      - name: Upload test report
+        uses: actions/upload-artifact@v4
         if: always()
         with:
-          name: playwright-report
+          name: playwright-report-shard-${{ matrix.shardIndex }}
           path: ./playwright-report/
           retention-days: 30
+
+  # Merge snapshots and commit
+  merge-and-commit:
+    needs: [setup, update-snapshots-sharded]
+    runs-on: ubuntu-latest
+    if: always()
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+        with:
+          ref: ${{ needs.setup.outputs.branch }}
+
+      # Download all snapshot artifacts from shards
+      - name: Download all snapshots
+        uses: actions/download-artifact@v4
+        with:
+          pattern: snapshots-shard-*
+          path: ./downloaded-snapshots
+          merge-multiple: false
+
+      # Merge snapshots from all shards into browser_tests directory
+      - name: Merge snapshots
+        run: |
+          echo "Merging snapshots from all shards..."
+          for shard_dir in ./downloaded-snapshots/snapshots-shard-*; do
+            if [ -d "$shard_dir" ]; then
+              echo "Copying from $shard_dir..."
+              cp -r "$shard_dir"/* browser_tests/ 2>/dev/null || true
+            fi
+          done
+          echo "Snapshot merge complete"
```

For some reason, this part worries me.
Can we fail loudly here instead of sending stderr and failed statuses to the cornfield?